larpandoracontent/LArControlFlow/StitchingCosmicRayMergingTool.cc

```diff
@@ -524,6 +524,49 @@ void StitchingCosmicRayMergingTool::StitchPfos(const MasterAlgorithm *const pAlg
         for (const ParticleFlowObject *const pPfo : pfoList) pfoVector.push_back(pPfo);
         std::sort(pfoVector.begin(), pfoVector.end(), LArPfoHelper::SortByNHits);
 
+        std::pair<const LArTPC*, const LArTPC*> stitchedLArTPCs;
+
+        try
+        {
+            for (const ParticleFlowObject *const pPfoToShift : pfoVector)
+            {
+                if (pfoToLArTPCMap.find(pPfoToShift) != pfoToLArTPCMap.end())
```

Comment:
Code is nice and clean by checking to see if pfoToShift is there, then using `.at(...)` below. I typically use `.count(...)` to check if something is there, as it reads a bit more cleanly than checking whether you got a valid iterator.

More efficient than doing the map lookup twice, though, is to actually store the iterator returned by `.find(...)` and then using it to get the LArTPC from the map too. i.e. only doing the look up once.

Probably doesn't matter much here, as the map is small, but when they get big this is important.